### DISCUSSION TITLE
Is there any conditional function to check if a query is from the query loop block?

### DISCUSSION THREAD

**Discussion by catkit** [WordPress/gutenberg#42655] [Q&A] [answered]
Hi there!

I'm working on a project where I need to filter `the_content()` that will be displayed by the query loop block only, I don't want to change `the_content()` when it's called inside other queries. So I was wondering if there's any conditional function like is_main_query() but to test if it's a query loop block.

Thanks!

**Comment by t-hamano** [comment, marked as answer]
Hi @catkit,

Sorry for the late reply.

I think what you want to do can be solved by the following approach:

- Apply a specific class to the post content block whose content you wish to rewrite.
- Use the [render_block_{blockname}](https://developer.wordpress.org/reference/hooks/render_block_this-name/) hook to rewrite content only if that class is present.

Here is a sample:

```php
add_filter( 'render_block_core/post-content', function ( $block_content, $block ) {
	// For the post content block that doesn't have a class name, output the content as is.
	if ( ! isset( $block['attrs']['className'] ) ) {
		return $block_content;
	}
	// For the post content block that doesn't have a specific class name, output the content as is.
	if ( 'your-class' !== $block['attrs']['className'] ) {
		return $block_content;
	}
	// For the post content block that has a specific class name, add contents.
	return $block_content . '<p>additional contents</p>';
}, 10, 2 );
```

**Comment by catkit** [comment]
Hi @t-hamano !

> Sorry for the late reply.

No worries! Thanks for taking the time to answer me!

I didn't know about this filter! I'll try it next time I need to change how a block is rendered!
